source/elements/oneDAL/source/data_management/data_sources/index.rst

```diff
@@ -0,0 +1,157 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
+=======
+Data Sources
+=======
+
+Data source is a concept of an out-of-memory storage for a :capterm:`dataset`.
+It is used at the data acquisition and data preparation stages for the
+following:
+
+- To extract datasets from external sources such as databases, files, remote
+  storage.
+
+- To load datasets into the device's local memory. Data do not always fit
+  the local memory, especially when processing with accelerators. A data source
+  provides the ability to load data by batches and extracts it directly into the
+  device's local memory. Therefore, a data source enables complex data analytics
+  scenarios, such as :capterm:`online computations <Online mode>`.
+
+- To filter and normalize :capterm:`feature` values that are being extracted.
+
+- To recover missing :capterm:`feature` values.
+
+- To detect :capterm:`outliers <outlier>` and recover the abnormal data.
+
+- To transform datasets into numerical representation. Data source shall
+  automatically transform non-numeric :capterm:`categorical <categorical
+  feature>` and :capterm:`continuous <continuous feature>` data values into one
+  of the numeric :capterm:`data formats <data format>`.
+
+CSV data source
+-----------------
+Class ``csv::data_source`` is an implementation of a data source type
```

Comment:
this is probably not implementation of data source type but api for accessing the data source represented as csv file. We need to explain the short name as well.
Also, keep in mind the existent documentation and find options to re-use that info here

Reply:
This was actually a placeholder copy-pasted from other place.
Now added few more details about csv data source.